test/end-to-end/test_launcher_restarts_supervisor.sh

```diff
@@ -8,28 +8,42 @@
 
 set -eou pipefail
 
+
+find_if_exists() {
+  command -v "${1}" || { log "Required utility '${1}' cannot be found!  Aborting."; exit 1; }
+}
+
+find_if_exists pgrep
+
 wait_for_sup_to_start() {
-	until pgrep hab-sup &>/dev/null; do
-		echo -n .
-		sleep 1
-	done
-	echo
+    until pgrep hab-sup &>/dev/null; do
+        echo -n .
+        sleep 1
+    done
+    echo
 }
 
 if pgrep hab-launch &>/dev/null; then
-	echo "Error: launcher process is already running"
-	exit 1
+    echo "Error: launcher process is already running"
+    exit 1
 fi
 
 TESTING_FS_ROOT=$(mktemp -d)
+echo "TESTING_FS_ROOT=${TESTING_FS_ROOT}"
 export TESTING_FS_ROOT
-sup_log="$TESTING_FS_ROOT/hab/sup/default/sup.log"
+#sup_log="$TESTING_FS_ROOT/hab/sup/default/sup.log"
+sup_log="sup.log"
 exit_file=$(mktemp)
 exit_code=${1:-1}
 
 mkdir -p "$(dirname "$sup_log")"
 echo -n "Starting launcher with root $TESTING_FS_ROOT (logging to $sup_log)..."
-env HAB_FEAT_TEST_EXIT="$exit_file" hab sup run &> "$sup_log" &
+
+
+# TODO: this doesn't appear to work properly. It'll download the new
+# launcher, but then somehow can't find the binary in the
+# TESTING_FS_ROOT. Unsure exactly what's going on yet.
```

Comment:
`TESTING_FS_ROOT` just doesn't work with actually installing packages. This test was originally written assuming the launcher hab been already installed.